.github/workflows/ci_linux_x64_clang_debug.yml

```diff
@@ -25,6 +25,7 @@ concurrency:
 
 jobs:
   setup:
+    if: ${{ github.repository_owner == 'iree-org' || github.event_name != 'schedule' }}
```

Comment:
We could also change jobs like this to configure their `runs-on` based on the repository owner, so if the workflow does run on `pull_request` triggers, it has a chance of succeeding without needing to add self-hosted runners to the repo.

```yml
# runs-on: azure-linux-scale
runs-on: ${{ github.repository_owner == 'iree-org' && 'azure-linux-scale' ||  'ubuntu-20.04'  }}
```

(Same comment for all workflows that have `pull_request` triggers.... except Windows right now)

Reply:
I think would work for `ci_linux_x64_clang_debug.yml` and `ci_linux_x64_clang_tsan.yml` but not for `ci_linux_arm64_clang.yml`? Generally I agree and that it would be a great improvement, but do you think we should do it in addition or instead of disabling running on schedule?

Reply:
Well, the failure mode for requesting a label that is not available in the repository is pretty poor: the workflow just waits until it times out. I think extending this pull request to also handle at least those Linux jobs that have the `pull_request` trigger makes sense.

Could either:

A) Keep current behavior (so PRs on forks would wait until GitHub's timeout, and the scheduled runs would similarly fail)
B) Skip unless the event is `workflow_dispatch` (sorta weird, would unbreak `pull_request` and `schedule` triggers but keep `workflow_dispatch` broken unless someone goes through the effort of registering a self-hosted running with the same label(s))
C) Switch runner based on the `github.repository_owner`
D) Switch runner based on the `github.repository_owner` _and_ also skip if the event is not workflow_dispatch
E) Switch runner based on the `github.repository_owner` _and_ also skip if the event is schedule (so they would run on pull_request but be slowwww)

I like option D or E.

Reply:
I switched it for some of the jobs where it makes sense. The runner is changed to a GH-hosted one only for those job where a run would be triggered by a workflow dispatch.